lib/fastly_nsq/message.rb

```diff
@@ -34,10 +34,17 @@ def finish
     nsq_message.finish
   end
 
-  def requeue(*args)
+  def requeue(timeout = nil)
     return managed if managed
+    timeout ||= requeue_period
 
     @managed = :requeued
-    nsq_message.requeue(*args)
+    nsq_message.requeue(timeout)
+  end
+
+  private
+
+  def requeue_period
+    ((attempts**4) + 45 + (rand(60) * (attempts + 1))) * 1_000
```

Comment:
I feel like this should have a `max` value...

Reply:
probably not a bad idea. By attempt 100 the requeue period is > 3 years (attempt 200 is > 50 years)

Reply:
capping attempts in the formula to 30 would max out at ~ 9.4 days

Reply:
`Nsq::Connection` has a similar formula

https://github.com/wistia/nsq-ruby/blob/c6ff6e506888067d3bdb3726799264bb9e7dfe4e/lib/nsq/connection.rb#L421-L427

They do limit the maximum sleep time.  You have some control over this indirectly by limiting `max_attempts`.

Reply:
https://github.com/fastly/fastly_nsq/pull/73